src/core/util/tools.py

```
@@ -23,18 +23,19 @@
 from src.core.constants import Constants
 
 
-def run_shell(shell: str) -> str:
+def run_shell(shell: str, log_ignore_regex: str = None) -> str:
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)